ext/pgsql/pgsql.c

```diff
@@ -4147,7 +4151,6 @@ PHP_FUNCTION(pg_flush)
 
 /* {{{ php_pgsql_meta_data
  * table_name must not be empty
- * TODO: Add meta_data cache for better performance
  */
 PHP_PGSQL_API zend_result php_pgsql_meta_data(PGconn *pg_link, const zend_string *table_name, zval *meta, bool extended)
```

Comment:
First let's drop the const qualifier, as the hash insertion is trying to increase the refcount of the zend_string (and also compute the hash on it).

```suggestion
PHP_PGSQL_API zend_result php_pgsql_meta_data(PGconn *pg_link, zend_string *table_name, zval *meta, bool extended)
```